pkg/api/controller.go

```diff
@@ -697,6 +704,25 @@ func (c *Controller) CreatePolicy(w http.ResponseWriter, r *http.Request, body C
 	ctx := r.Context()
 	c.LogAction(ctx, "create_policy", r, "", "", "")
 
+	// Check that policy ID is valid
+	valid, msg := c.isNameValid(body.Id, "Policy")
+	if !valid {
+		writeError(w, http.StatusBadRequest, msg)
+		return
+	}
+	
+	// Check that we don't already have a policy with this ID
+	_, gpErr := c.Auth.GetPolicy(ctx, body.Id)
+	// if err is nil, it means we got a policy, which means we can't create
+	// a new policy with this ID. Return a 409 Conflict http status code
+	// which means the request conflicts with the current state of the server
+	if gpErr == nil {
+		gpErr = auth.ErrAlreadyExists
+		if c.handleAPIError(ctx, w, gpErr) {
+			return
+		}
+	}
```

Comment:
AFAICT this is a race condition.  And one that can actually occur, e.g. if my GUI is stuck and I redo a similar operation from another window, when both are unstuck we can race here.

We should use some SetIf capability on the database, as part of the `c.Auth.WritePolicy` call.